ports/sail/portfile.cmake

```
@@ -1,67 +1,80 @@
-    REPO HappySeaFox/sail
-    REF "v${VERSION}"
-    SHA512 755e028b0cb0a2bb3ec97bfbc888a8afbf73923a36da2a0d32c535bd268c8e087404f44f233a6ef186234df108232d9315d71b7efd9abd2f088d988d4360320d
-    HEAD_REF master
-    PATCHES
-        fix-include-directory.patch
-)
-
-# Enable selected codecs
-set(ONLY_CODECS "")
-
-foreach(CODEC avif bmp gif ico jpeg jpeg2000 jpegxl pcx png psd qoi svg tga tiff wal webp xbm)
-    if (${CODEC} IN_LIST FEATURES)
-        list(APPEND ONLY_CODECS ${CODEC})
-    endif()
-endforeach()
-
-list(JOIN ONLY_CODECS "\;" ONLY_CODECS_ESCAPED)
-
-# Enable OpenMP
-if ("openmp" IN_LIST FEATURES)
-    set(SAIL_ENABLE_OPENMP ON)
-endif()
-
-vcpkg_cmake_configure(
-    SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS
-        -DBUILD_TESTING=OFF
-        -DSAIL_COMBINE_CODECS=ON
-        -DSAIL_ENABLE_OPENMP=${SAIL_ENABLE_OPENMP}
-        -DSAIL_ONLY_CODECS=${ONLY_CODECS_ESCAPED}
-        -DSAIL_BUILD_APPS=OFF
-        -DSAIL_BUILD_EXAMPLES=OFF
-)
-
-vcpkg_cmake_install()
-
-vcpkg_copy_pdbs()
-
-# Remove duplicate files
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include"
-                    "${CURRENT_PACKAGES_DIR}/debug/share")
-
-# Move cmake configs
-vcpkg_cmake_config_fixup(PACKAGE_NAME sail       CONFIG_PATH lib/cmake/sail       DO_NOT_DELETE_PARENT_CONFIG_PATH)
-vcpkg_cmake_config_fixup(PACKAGE_NAME sailcodecs CONFIG_PATH lib/cmake/sailcodecs DO_NOT_DELETE_PARENT_CONFIG_PATH)
-vcpkg_cmake_config_fixup(PACKAGE_NAME sailcommon CONFIG_PATH lib/cmake/sailcommon DO_NOT_DELETE_PARENT_CONFIG_PATH)
-vcpkg_cmake_config_fixup(PACKAGE_NAME sailc++    CONFIG_PATH lib/cmake/sailc++    DO_NOT_DELETE_PARENT_CONFIG_PATH)
-vcpkg_cmake_config_fixup(PACKAGE_NAME sailmanip  CONFIG_PATH lib/cmake/sailmanip  DO_NOT_DELETE_PARENT_CONFIG_PATH)
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake"
-                    "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")
-
-
-# Fix pkg-config files
-vcpkg_fixup_pkgconfig()
-
-# Unused because SAIL_COMBINE_CODECS is ON, removes an absolute path from the output
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/sail-common/config.h" "#define SAIL_CODECS_PATH [^\r\n]+[\r\n]*" "" REGEX)
-
-# Handle usage
-file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-
-# Handle copyright
-vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO HappySeaFox/sail
+    REF "v${VERSION}"
+    SHA512 0e6bd4fb7910eda29e0cd6d96e31ff9a476d5836055e4653d6401ab72209eccd3624b8c1e92b7cd0d22ecdaa8ffde44b155da36da61c99ec0c06e1d388bd1d67
+    HEAD_REF master
+    PATCHES
+        fix-heif.patch
+        fix-include-directory.patch
+)
+
+# Enable selected codecs
+set(ONLY_CODECS "")
+
+# The list is sorted by priority, with each priority group separated by triple space
+foreach(CODEC gif jpeg png svg webp   avif ico   heif openexr psd tiff   bmp hdr jpeg2000 jpegxl pnm qoi tga   jbig pcx wal xbm xpm xwd)
```

I don't think priority is really useful here, but...
```suggestion
set(highest_priority_codes gif jpeg png svg webp)
set(secondary_codecs  ...)
foreach(CODEC IN LISTS highest_priority_codes secondary_codecs)
```